### DISCUSSION TITLE
How to prevent Error: Cannot find module 'semver/functions/satisfies' and run npm properly

### DISCUSSION THREAD

**Discussion by hgftrdw45ud67is8o89** [nodejs/help#4690] [Q&A]
```
set PATH=G:\llm\Sillytavern_fullset\nodejs;.........................................
\nodejs\node_modules\npm\lib\es6\validate-engines.js:31
    throw err
    ^

Error: Cannot find module 'semver/functions/satisfies'
Require stack:
- G:\llm\Sillytavern_fullset\nodejs\node_modules\npm\lib\cli-entry.js
- G:\llm\Sillytavern_fullset\nodejs\node_modules\npm\lib\cli.js
- G:\llm\Sillytavern_fullset\nodejs\node_modules\npm\bin\npm-cli.js

```
I know SO has answered it ,but it doesnt seems to work? As shown above I added the top nodejs folder into path.

Is there anything else I need to do? I tried adding the nodemodule\ to path but doesn't seems to work either.

I downloaded the zip unzip and renamed it to nodejs.

**Comment by shwetankg07** [comment]
For anyone landing here with the same error: this isn't a PATH problem. npm ships with its own copies of the libraries it needs, inside `nodejs\node_modules\npm\node_modules\`, and `semver` is one of them. So "Cannot find module 'semver/functions/satisfies'" coming from `npm\lib\...` means npm's own files are missing, and that almost always means the zip didn't extract completely.

Check whether the file is there:

```bat
dir G:\llm\Sillytavern_fullset\nodejs\node_modules\npm\node_modules\semver\functions\satisfies.js
```

If it says "File Not Found", the extraction was incomplete. Delete the `nodejs` folder and extract the zip again. 7-Zip works, and so does the `tar` that ships with Windows 10 and 11:

```bat
tar -xf node-v20.x.x-win-x64.zip
```

(Rename the extracted folder to `nodejs` again afterwards.) If files still go missing, check whether your antivirus quarantined anything from that folder.

Then `node -v` and `npm -v` should both work.
